go/client/cas_test.go

```diff
@@ -625,3 +628,158 @@ func TestFlattenActionOutputs(t *testing.T) {
 		}
 	}
 }
+
+func TestDownloadActionOutputs(t *testing.T) {
+	ctx := context.Background()
+	listener, err := net.Listen("tcp", ":0")
+	if err != nil {
+		t.Fatalf("Cannot listen: %v", err)
+	}
+	defer listener.Close()
+	server := grpc.NewServer()
+	fake := fakes.NewCAS()
+	bsgrpc.RegisterByteStreamServer(server, fake)
+	regrpc.RegisterContentAddressableStorageServer(server, fake)
+	go server.Serve(listener)
+	defer server.Stop()
+	c, err := client.Dial(ctx, instance, client.DialParams{
+		Service:    listener.Addr().String(),
+		NoSecurity: true,
+	}, client.UseBatchOps(true))
+	if err != nil {
+		t.Fatalf("Error connecting to server: %v", err)
+	}
+	defer c.Close()
+
+	fooDigest := fake.Put([]byte("foo"))
+	barDigest := fake.Put([]byte("bar"))
+	dirB := &repb.Directory{
```

Comment:
nit: dirA first?

Reply:
Hmm, I wonder how that even happened...

Reply:
Oh, that's how -- note that dirA includes dirB. So it needs to be defined second.